### PR TITLE
Release candidate 1

### PR DESCRIPTION
**Pull request by prydin**
The code is feature complete and has undergone a reasonable amount of testing.

The README file has a few TODOs in it regarding links to releases and Docker repo name. I need help resolving those. 

To build a release, simply type ```make release``` and you will find all binaries under "target" and it will also build you a Docker image.

### CODE REVIEW

**Comment by prydin** [comment]
@vikramraman I noticed this got assigned to me. Are you about to finish up the review? The PR is still blocked pending review.

**Comment by vikramraman** [comment]
@prydin I will finish up the review later today.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	$(GOTEST) -v ./...
+clean: 
+	$(GOCLEAN)
+	rm -f $(BINARY_NAME)
+	rm -rf $(TARGET)
+run:
+	$(GOBUILD) -o $(BINARY_NAME) -v ./...
+	./$(BINARY_NAME)
+deps:
+	$(GOGET) github.com/wavefrontHQ/prometheus-storage-adapter
```
Comment (vikramraman) [inline Makefile]: Let's use lowercase `wavefronthq` throughout (Makefile, README and go files) for consistency as we use lowercase for all the other go pkgs under wavefrontHQ.

Comment (prydin) [reply Makefile]: I don't think that's possible, since that's what it's named in github. If you go a "go get", you're going to get names ending with HQ.

Comment (vikramraman) [inline Makefile]: Also goget `github.com/Sirupsen/logrus` and `github.com/prometheus/prometheus` and any other thirdparty dependencies.

Comment (prydin) [reply Makefile]: Shouldn't be needed. Go get should pull them down by analyzing the imports in our source files. I tried to build this on a fresh VM and it pulled down the libs correctly.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+GOGET=$(GOCMD) get
+DOCKER=docker
+BINARY_NAME=adapter
+TARGET=target
+BINARY_LINUX=$(TARGET)/$(BINARY_NAME)_linux
+BINARY_DARWIN=$(TARGET)/$(BINARY_NAME)_darwin
+BINARY_WINDOWS=$(TARGET)/$(BINARY_NAME)_windows.EXE
+
+all: build test
+build:
```
Comment (vikramraman) [inline Makefile]: Perhaps `build:deps` ?

Comment (prydin) [reply Makefile]: In my opinion, you want ```make build``` to be as fast as possible when you're in a edit-build-debug cycle, so it's often considered best practice to keep deps out of the build target.

HOWEVER, the "all" target should definitely have a reference to deps. Will fix that.

File: Makefile (hunk 1 above)
Comment (vikramraman) [reply Makefile]: Okay adding `build:deps` should fix it then. It was failing for me as `build` (default Makefile target) doesn't do Goget currently.

Comment (vikramraman) [reply Makefile]: `go get github.com/wavefronthq/prometheus-storage-adapter` works.

Comment (prydin) [reply Makefile]: OK. The readme should have a reference to "deps". Will fix that.

Comment (prydin) [reply Makefile]: ...actually, it already has. The readme file says ```make deps build```

File: adapter.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/wavefrontHQ/prometheus-storage-adapter/backend"
+
+	log "github.com/Sirupsen/logrus"
```
Comment (vikramraman) [inline adapter.go]: From https://github.com/sirupsen/logrus/blob/master/README.md:

```
Everything using logrus will need to use the lower-case: github.com/sirupsen/logrus. Any package that isn't, should be changed.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,63 @@
...
+### Example
+To run the adapter listening to port 1234 and sending results to localhost:2878, we can use the following command. This command also adds a prefix ("prom") to all metrics coming from the adapter.
+```
+./adapter -proxy localhost -proxy-port 2878 -listen 1234 -prefix prom
+```
+
+## Running the adapter in a Docker container
+The adapter is available as a Docker image. To run it, simply type ```docker run <TODO: Name of repo>``` with the parameters discussed above, for example:
+```
+docker run <TODO: Name of repo> -proxy localhost -proxy-port 2878 -listen 1234 -prefix prom
```
Comment (vikramraman) [inline README.md]: You can use `wavefronthq/prometheus-storage-adapter` for the repo (https://hub.docker.com/r/wavefronthq/prometheus-storage-adapter/). Will push a image once PR is complete.

File: adapter.go (hunk 3 above)
Comment (prydin) [reply adapter.go]: Fixed.

File: README.md (hunk 4 above)
Comment (prydin) [reply README.md]: Fixed

File: Makefile (hunk 2 above)
Comment (prydin) [reply Makefile]: Fixed

File: Makefile (hunk 1 above)
Comment (prydin) [reply Makefile]: Fixed

Comment (prydin) [reply Makefile]: You are absolutely right. My bad. Fixed.

**Review by vikramraman — APPROVED** [review]

**Pull request merged**